layers/cmd/valueset.go

```diff
@@ -211,7 +216,40 @@ func (vsets Valuesets) Lookup(ctx *ls.Context, req ls.ValuesetLookupRequest) (ls
 		ctx.GetLogger().Debug(map[string]interface{}{"valueset.found": found})
 		return found, nil
 	}
+	var wg sync.WaitGroup
+	wg.Add(len(req.TableIDs))
+	vsr := make(chan ls.ValuesetLookupResponse, 1)
```

Comment:
You are trying to workaround deadlock with channel of 1. Best synchronization is no synchronization.

Create result:=make([]ls.ValuesetLookupResponse,n) and err:=make([]error,n), and start the goroutines with index i. Each goroutine writes to result[i] and err[i]. This is deadlock free then. Waitgroup is still required.